pkg/trace/local_tracer.go

```diff
@@ -85,46 +86,57 @@ func NewLocalClient(cfg *config.Config, logger log.Logger, chainID, nodeID strin
 		logger:  logger,
 	}
 
-	go lc.draincanal()
+	go lt.drainCanal()
+	if cfg.Instrumentation.TracePullAddress != "" {
+		go lt.servePullData()
+	}
+	if cfg.Instrumentation.TracePushConfig != "" {
+		s3Config, err := readS3Config(path.Join(cfg.RootDir, "config", cfg.Instrumentation.TracePushConfig))
+		if err != nil {
+			return nil, fmt.Errorf("failed to read s3 config: %w", err)
+		}
+		lt.s3Config = s3Config
+		go lt.pushLoop()
+	}
```

Comment:
I don't like this, very open to better ideas for managing pushing and pulling. used a separate config (:sob:) for the s3 cause adding a bunch of s3 specific configurations seems like a bad idea. maybe we should just force the push config be included in the config directory and then turn this into a bool to make it less confusing